### ISSUE TITLE
No Web UI

### ISSUE THREAD

**Issue by gijsje** [EliasKotlyar/Xiaomi-Dafang-Hacks#369]
I have done all to the letter but i can not get it to work.
I have the camera connected to the wifi as i can see DAFANG with the ip address.
If i go to the ip adres in the browser nothing is shown.
I can connect to the camera over FTP and with putty.

What could be the issue???

Thanks
Gijs

**Comment by jmtatsch** [comment]
Possibly the enabled ssl module. Can you post your startup log?

**Comment by gijsje** [comment]
Here is the startlog

==================================================
Starting the Dafang Hacks Custom Application Layer
==================================================
loaded common functions
Bind mounted /system/sdcard/root to /root
Bind mounted /system/sdcard/etc to /etc
wpa_supplicant: Successfully initialized wpa_supplicant
rfkill: Cannot open RFKILL control device
nl80211: Could not re-add multicast membership for vendor events: -2 (No such file or directory)
udhcpc: udhcpc (v1.22.1) started
Sending discover...
Sending discover...
Sending select for 192.168.0.163...
Lease of 192.168.0.163 obtained, lease time 3600
deleting routers
adding dns 8.8.8.8
adding dns 8.8.4.4
initialized gpios
dropbear: 
bftpd: 
created new certificate for webserver
lighttpd: 
Startup finished!

**Comment by franciz16** [comment]
Same issue here.
I can see Cam on my Wifi, ping is O.K.
Yellow led flashing all the time ... is this O.K.?

**Comment by jmtatsch** [comment]
@gijsje log looks good. Can you try https? Also try rebooting once

**Comment by gijsje** [comment]
@jmtatsch no luck with https
Reboot does not matter

**Comment by Deltaex** [comment]
Same for me, it worked yesterday.
I reset everything today and no more access to www

**Comment by gijsje** [comment]
If i change
ssl.engine = "enable"
ssl.pemfile = "/system/sdcard/config/lighttpd.pem"

to 
#  ssl.engine = "enable"
#  ssl.pemfile = "/system/sdcard/config/lighttpd.pem"

it works

**Comment by jmtatsch** [comment]
Hmm, but /system/sdcard/config/lighttpd.pem on camera is non-empty?

**Comment by gijsje** [comment]
There is no file lighttpd.pem only lighttpd.conf and lighttpd.user

**Comment by huntz** [comment]
Hi,
starting the lighttpd by hand, I get this error:

```
/system/sdcard/bin/lighttpd -f /system/sdcard/config/lighttpd.conf
2018-05-20 03:11:49: (mod_openssl.c.405) SSL: BIO_read_filename('/system/sdcard/config/lighttpd.pem') failed
2018-05-20 03:11:49: (server.c.1148) Initialization of plugins failed. Going down.
```

The file /system/sdcard/config/lighttpd.pem is obviously missing.

If I run this command, to generate the file, the webserver is starting correctly:
```
openssl req -new -x509 -keyout /system/sdcard/config/lighttpd.pem -out /system/sdcard/config/lighttpd.pem -days 365 -nodes -subj "/C=DE/ST=Bavaria/L=Munich/O=.../OU=.../CN=.../emailAddress=..."
```

So, the problem is in run.sh, generating the lighttpd.pem.

**Comment by keirans** [comment]
Confirmed here as well.
I had to set the OPENSSL_CONF variable to get this working also.

```
[root@DAFANG:config]# openssl req -new -x509 -keyout /system/sdcard/config/lighttpd.pem -out /system/sdcard/config/lighttpd.pem -days 365 -nodes -subj "/C=DE/ST=Bavaria/L=Munich/O=.../O
U=.../CN=.../emailAddress=..."
WARNING: can't open config file: /home/osboxes/SRC/Xiaomi-Dafang-Software/libressl/_install/etc/ssl/openssl.cnf
Unable to load config info from /home/osboxes/SRC/Xiaomi-Dafang-Software/libressl/_install/etc/ssl/openssl.cnf
[root@DAFANG:config]# find / -name openssl.cnf
[root@DAFANG:config]# export OPENSSL_CONF=/system/sdcard/config/openssl.cnf
[root@DAFANG:config]# openssl req -new -x509 -keyout /system/sdcard/config/lighttpd.pem -out /system/sdcard/config/lighttpd.pem -days 365 -nodes -subj "/C=DE/ST=Bavaria/L=Munich/O=.../O
U=.../CN=.../emailAddress=..."
Generating a 1024 bit RSA private key
......................++++++
...........++++++
writing new private key to '/system/sdcard/config/lighttpd.pem'
-----
[root@DAFANG:config]# /system/sdcard/bin/lighttpd -f /system/sdcard/config/lighttpd.conf
[root@DAFANG:config]#
```

**Comment by jmtatsch** [comment]
I really don't understand why the generation of the certificate in run.sh doesn't work reliably.
Until then ssl support is commented out again in 91e39cd399bb5866b15fb38c15dc38bb379efa6f.

**Comment by jmtatsch** [comment]
I fixed the problem in 95a6b32d2d9ee9d8d705c7f6b34137a52c2bc3f6.
Stupid me wasn't using the openssl from /system/sdcard/bin to generate the certificates.
